framework/src/meshgenerators/FancyExtruderGenerator.C

```diff
@@ -510,6 +577,33 @@ FancyExtruderGenerator::generate()
 
         new_elem = mesh->add_elem(new_elem);
 
+        // maintain extra integers
+        if (num_extra_elem_integers > 0)
+        {
+          for (unsigned int i = 0; i < num_extra_elem_integers; i++)
+          {
+            // declare the "control_drum_id" when the first elem is added
+            if (!mesh->has_elem_integer(id_names[i]))
+              mesh->add_elem_integer(id_names[i]);
```

Comment:
Can you add all of the integers that don't exist up front instead of within the element loop? Also - remove the comment that's hanging around